port/protobuf.h

```diff
@@ -62,7 +62,7 @@ namespace protobuf = google::protobuf;
 
 inline bool RequiresUtf8Validation(
     const google::protobuf::FieldDescriptor& descriptor) {
-  // commit d8c2501b43c1b56e3efa74048a18f8ce06ba07fe of >= v3.22.0
+  // commit d85c9944c55fb38f4eae149979a0f680ea125ecb of >= v3(!).22.0
```

Comment:
why (!)?

Reply:
@vitalybuka because the number we comparse against is 4022000 starting with 4 despite 3.x.x rather than 4.x.x.

Reply:
I propose, in future PR, drop all `// commit bb30225f06c36399757dc698b409d5f79738e8d1 of >=3.12.0`
and add comment with set of instructions how to calculate constants like 3021005?